crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -683,6 +693,79 @@ impl EventCache {
             }
         }
     }
+
+    #[cfg(feature = "experimental-search")]
+    fn parse_timeline_event(event: &TimelineEvent) -> Option<AnySyncMessageLikeEvent> {
```

Comment:
Can you make this a freestanding function? It's weird that this is a method attached to the `EventCache` (and it was already a bit surprising to have it in the previous place where it was).